### PR TITLE
Add tfa.weightNormalization

### PR DESCRIPTION
**Pull request by humblemat810**
weight normalization has been proposed in
https://arxiv.org/abs/1602.07868
and this implement the paper proposed idea

advanced activation is also supported by setting:
activation = 'linear'
activation_layer = tensorflow.keras.layers.leakyRELU

### CODE REVIEW

**Comment by philipperemy** [comment]
@humblemat810 looks good. Any reasons why you removed the tests for tensorflow 1.15?

**Comment by humblemat810** [comment]
> @humblemat810 looks good. Any reasons why you removed the tests for tensorflow 1.15?

The only reason is, the tensorflow addon that has weight normalisation implemented, requires tensorflow 2.2
I manually tested to load weight norm on tf 2.1 and the required module already not exist in 2.1, so the test on 1.5 must fail
tldr : the tfa weight norm requires tf 2.2

**Comment by philipperemy** [comment]
Makes sense then.

**Comment by humblemat810** [comment]
still one more problem... there is No test case testing on the validity of weight norm, I think no test case so far enabled weight norm

**Comment by philipperemy** [comment]
@humblemat810 yes please don't hesitate to show that no regressions will occur with this PR and that the default behavior is unchanged.

**Comment by philipperemy** [comment]
I'll test this PR soon. I hope to merge it once my GPU server is backed online.

**Comment by philipperemy** [comment]
It has been added thank you!!
